agent/src/beerocks/fronthaul_manager/fronthaul_manager_main.cpp

```diff
@@ -204,8 +205,24 @@ int main(int argc, char *argv[])
     std::string agent_uds =
         beerocks_slave_conf.temp_path + std::string(BEEROCKS_SLAVE_UDS) + "_" + fronthaul_iface;
 
+    // Find correct interface number for get correct path for hostap_ctrl
+    beerocks::bpl::BPL_WLAN_IFACE interfaces[beerocks::IRE_MAX_SLAVES] = {0};
+    int num_of_interfaces                                              = beerocks::IRE_MAX_SLAVES;
+    if (beerocks::bpl::cfg_get_all_prplmesh_wifi_interfaces(interfaces, &num_of_interfaces)) {
+        std::cout << "failed to read interfaces map" << std::endl;
+        return 1;
+    }
+
+    uint8_t iface_num = 0;
+    for (iface_num = 0; iface_num < beerocks::IRE_MAX_SLAVES; iface_num++) {
+        if (interfaces[iface_num].ifname == fronthaul_iface) {
+            break;
+        }
+    }
+
     // Create ap_manager
-    son::ap_manager_thread ap_manager(agent_uds, fronthaul_iface, *g_logger_ap_mananger);
+    son::ap_manager_thread ap_manager(agent_uds, fronthaul_iface, *g_logger_ap_mananger,
+                                      beerocks_slave_conf.hostap_ctrl_iface[iface_num]);
```

Comment:
I made a comment earlier that this path should be retrieved directly from bpl. Actually, this is a better place for it. I.e.,
```suggestion
                                      bpl_get_hostap_ctrl_iface(fronthaul_iface));
```

The UCI implementation can be pretty simple: it just returns "/var/run/hostapd/" + fronthaul_iface, just like the current code. That's OK because UCI means OpenWrt and there the path is always that. And the linux implementation can read it from the platform_db file.

Note that this means that all the previous commits become redundant, except the first one. You still need the first one to populate the default platform_db. Note that we don't have a prplmesh_db.in file, it has to be set from prplmesh_utils.sh.in.